src/node/nodeHttpClientSession.ts

```diff
@@ -61,7 +61,7 @@ export class NodeHttpClientSession implements IHttpClientSession {
           : Http.request(this.requestOptions(url, request), handler);
 
       rawRequest.on("error", error => {
-        reject(error);
+        reject(new Error(`Request to ${url.toString()} failed: ${error.message}`));
```

Comment:
Are you sure there's always `message` field on error? Maybe there's, I just frankly don't know :)
But if in doubt, maybe let's do something like `${error.message != null ? error.message : error}`. What do you think?